pkg/compactor/block_upload.go

```diff
@@ -528,6 +542,32 @@ func (c *MultitenantCompactor) validateBlock(ctx context.Context, blockID ulid.U
 	return nil
 }
 
+func (c *MultitenantCompactor) validateMaximumBlockSize(logger log.Logger, files []metadata.File, userID string) error {
+	maxBlockSizeBytes := c.cfgProvider.CompactorBlockUploadMaxBlockSizeBytes(userID)
+	if maxBlockSizeBytes <= 0 {
+		return nil
+	}
```

Comment:
We already check for this on lines 411-413.

Reply:
I'm not following what you mean here. If there's no limit to enforce we don't need to do anything else in this function. Did you mean to comment on the negative file size check? That's for correctness within the function itself to make the overflow check clear.